src/components/SearchForm.js

```diff
@@ -58,21 +59,35 @@ export default class SearchForm extends React.Component {
      * @param {Object} response
      */
     handleResponse(response) {
-        // TODO: handle CourseItem
-        const docs = response.data[1].response.docs;
-        const highlights = response.data[1].highlighting;
-        let results = docs.map((doc, index) => {
-            console.log(doc.description != '' ? doc.description : highlights[doc.id].content[0])
-            console.log(doc.name)
-            console.log(doc.siteName != '' ? doc.siteName : '')
-            return {
-                url: doc.id,
-                description: doc.description != '' ? doc.description : highlights[doc.id].content[0],
-                pageName: doc.name,
-                siteName: doc.siteName != '' ? doc.siteName : ''
-            }
-        });
+        let results = [];
+        for (var i = 0; i < response.data.length; i++) {
+            const dataset = response.data[i];
+            const docs = dataset.response.docs;
+            const highlights = dataset.highlighting;
+            switch (dataset.type) {
+                case "genericPage":
+                    results = results.concat(docs.map(doc => {
+                        return {
+                            url: doc.id,
+                            description: doc.description != '' ? doc.description : highlights[doc.id].content[0],
+                            pageName: doc.name,
+                            siteName: doc.siteName != '' ? doc.siteName : ''
+                        }
+                    }));
+                    break;
 
+                case "courseItem":
+                    results = results.concat(docs.map(doc => {
+                        return {
+                            url: doc.id,
+                            description: doc.description,
+                            pageName: doc.name,
+                            siteName: doc.subjectData.length == 2 ? doc.subjectData[1] : ''
+                        }
+                    }))
+                    break;
+            }
+        }
```

Comment:
Nice work here! One minor thing I would suggest is just doing `results = docs.map(...`, no need to `concat` because there's nothing in the array. You can leave the declaration as `let results = [];` since that's what results should be by default.

Reply:
@bfbachmann Isn't `concat` needed to make sure that with multiple cores being returned, all the results from each core get appended to `results`?

Reply:
Yup you're right, sorry! I didn't notice the `for` loop 😬 .